hydrolib/core/io/dflowfm/bc/models.py

```diff
@@ -566,6 +593,20 @@ def _validate_quantityunitpairs(cls, values: Dict) -> Dict:
             else len(verticalpositions.split())
         )
 
+        return number_of_verticalpositions
+
+    @root_validator(pre=True)
+    def _validate_quantityunitpairs(cls, values: Dict) -> Dict:
+        super().validate_and_update_quantityunitpairs(values)
```

Comment:
Isn't this now called twice? Because in `VectorForcingBase` is it also a `@root_validator`. Maybe check in debug mode with a breakpoint.

Reply:
Indeed! Let's say we create a `T3D`, then `ForcingBase._validate_quantityunitpair` is called, because it's a `root_validator` for the `ForcingBase`. Next, we call the `root_validator` in `VectorForcingBase`, where we call the same `ForcingBase._validate_quantityunitpair` again. Finally, the `root_validator` in the `T3D` class calls `VectorForcingBase.validate_and_update_quantityunitpairs` again, which in turn calls `ForcingBase._validate_quantityunitpair` a third time.

I will remove the calls to the base validators, as they are already called automatically.

Good catch! :)

Reply:
Done!